source/common/http/conn_manager_impl.h

```diff
@@ -632,6 +634,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     // response.
     bool encoding_headers_only_{};
     Network::Socket::OptionsSharedPtr upstream_options_;
+    Tracing::CustomTagMap tracing_custom_tags_;
```

Comment:
perf nit: In the common case this won't be used. Can you do something similar to `request_metadata_map_vector_` where by default it's a unique_ptr and we lazy initialize it?